### PR TITLE
AlphonsoCode/DEF#2328. ignore error from datanode in readFile

### PR DESCRIPTION
**Pull request by aseem-alphonso**

### CODE REVIEW

File: lib/webhdfs.js [hunk 1]
```diff
@@ -520,7 +520,7 @@ WebHDFS.prototype.readFile = function readFile (path, opts, callback) {
   var hdfsWatchdog = setTimeout(timeoutOperation, timeoutDuration);
 
   remoteFileStream.once('error', function onError (err) {
-    error = err;
+    if(err && err.port !== 50075) error = err;
```
Comment (tushar-rishav) [inline lib/webhdfs.js]: We should also check if the error code is `ECONNREFUSED`. You don't want to ignore legit errors thrown on 50075 port by `createReadStream`. For instance from here https://github.com/AlphonsoCode/webhdfs/blob/51b8d09fff41ad48ec607033181a2eebe7e099ad/lib/webhdfs.js#L728-L731

**Review by tushar-rishav — CHANGES_REQUESTED** [review]
Check error code as well. Also, required to update the commit author.

File: lib/webhdfs.js (hunk 1 above)
Comment (aseem-alphonso) [reply lib/webhdfs.js]: done.

**Review by tushar-rishav — APPROVED** [review]
LGTM. Thanks

**Pull request merged**